cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -458,6 +494,26 @@ buildPoolParams nw dir index specifiedRelays = do
    poolVrfVKF = File $ dir </> "vrf" ++ strIndex ++ ".vkey"
    poolRewardVKF = File $ dir </> "staking-reward" ++ strIndex ++ ".vkey"
 
+computeDelegation
+  :: NetworkId
+  -> FilePath
+  -> Ledger.PoolParams StandardCrypto
+  -> ExceptT GenesisCmdError IO Delegation
+computeDelegation nw delegDir dPoolParams = do
+    payVK <- firstExceptT GenesisCmdFileInputDecodeError
+                  $ newExceptT $ Keys.readVerificationKeyOrFile AsPaymentKey (VerificationKeyFilePath payVKF)
+    stakeVK <- firstExceptT GenesisCmdFileInputDecodeError
```

Comment:
can be joined in a single `ExceptT`

Reply:
I went for some code sharing instead, to make this more digestible:

```diff
-    payVK <- firstExceptT GenesisCmdFileInputDecodeError
-                  $ newExceptT $ Keys.readVerificationKeyOrFile AsPaymentKey (VerificationKeyFilePath payVKF)
-    stakeVK <- firstExceptT GenesisCmdFileInputDecodeError
-                  $ newExceptT $ Keys.readVerificationKeyOrFile AsStakeKey (VerificationKeyFilePath stakeVKF)
+    payVK   <- readVKeyFromDisk AsPaymentKey payVKF
+    stakeVK <- readVKeyFromDisk AsStakeKey   stakeVKF
     let paymentCredential = PaymentCredentialByKey $ verificationKeyHash payVK
         stakeAddrRef = StakeAddressByValue $ StakeCredentialByKey $ verificationKeyHash stakeVK
         dInitialUtxoAddr = makeShelleyAddressInEra ShelleyBasedEraShelley nw paymentCredential stakeAddrRef
@@ -514,6 +512,9 @@ computeDelegation nw delegDir dPoolParams = do
    where
      payVKF = File @(VerificationKey ()) $ delegDir </> "payment.vkey"
      stakeVKF = File @(VerificationKey ()) $ delegDir </> "staking.vkey"
+     readVKeyFromDisk role file =
+      firstExceptT GenesisCmdFileInputDecodeError $ newExceptT $
+        Keys.readVerificationKeyOrFile role (VerificationKeyFilePath file)
```